wallet/src/main/scala/org/bitcoins/wallet/LockedWallet.scala

```diff
@@ -110,7 +119,8 @@ abstract class LockedWallet
                     falsePositiveRate = walletConfig.bloomFalsePositiveRate,
                     flags = bloomFlag)
 
-      pubkeys.foldLeft(baseBloom) { _.insert(_) }
+      val withPubs = pubkeys.foldLeft(baseBloom) { _.insert(_) }
+      outpoints.foldLeft(withPubs) { _.insert(_) }
```

Comment:
Is there any reason we can't have some kind of persistent bloom filter lying around so that we don't have to reconstruct the whole thing each time we add a new outpoint? (I understand that it needs to be reconstructed from scratch if anything ever gets removed, but for insertions this should be constant time not linear/quadratic.)

Reply:
I guess that would be an improvement, but that sounds like introducing a lot of complexity for a very limited gain. Here's a quick benchmark of inserting 10K pubkeys and outpoints: 

```scala
    val pubkeys = (1 to 10000).map(_ => CryptoGenerators.publicKey.sampleSome)
    val outpoints =
      (1 to 10000).map(_ => TransactionGenerators.outPoint.sampleSome)

    val start = Deadline.now
    val baseBloom = BloomFilter(30000, 0.001, BloomUpdateAll)

    val end = Deadline.now
    val withPubs = pubkeys.foldLeft(baseBloom) { _.insert(_) }
    val withPoints = outpoints.foldLeft(withPubs) { _.insert(_) }
    println(s"took time: ${(end - start).toMillis} millis")
```

When executing this with bloop, I get `took time: 20 millis` (the actual execution takes  a lot longer, but that's spent on generating pubkeys and outpoints). So I don't think this is a big issue. Unless you have a nice way of implementing it